### PR TITLE
Remove small OCaml3 leftovers

### PR DESCRIPTION
**Pull request by ghost**
tl;dr: this removes long dead code in the native code generator.

Long version:

In the native code generator, `Proc.word_addressed` was used to prefer loading smaller-than-word memory data as whole words and then appropriate masking, rather than byte or short loads.

This was required by the alpha backend since, until the introduction of the BWX extension in 21164a processors, this processor lacked byte and short memory access instructions.

But the alpha backend went the way of the dodo after OCaml 3, and unless there are plans to support TI's C35 family of DSP, one can reasonably expect OCaml won't have to run on processors lacking such instructions, even future processors - the alpha experience has taught CPU designers that this was not that a wise idea.

### CODE REVIEW

**Review by nojb — APPROVED** [review]
Clearly correct. An approval from @xavierleroy would be nice though.

**Comment by ghost** [comment]
Updated after running `make alldepend`.

**Comment by shindere** [comment]
I very much like this PR! Thanks both for the digging and the
explanations of the context, which are very valuable for someone not
aware of these things.

I did take the time to go through the diff and ensure that there are now
no left occurrences of `word_addressed` in the codebase.

**Review by shindere — APPROVED** [review]

**Comment by ghost** [comment]
> I did take the time to go through the diff and ensure that there are now no left occurrences of `word_addressed` in the codebase.

That's `git grep`'s job :)

**Comment by shindere** [comment]
Miod Vallat (2024/03/22 03:24 -0700):
> That's `git grep`'s job :)

That's what I used :) I indeed didn't litterally read all the files...
;)

**Comment by Octachron** [comment]
Merged (after checking offline that @xavierleroy was fine with the change).

**Comment by xavierleroy** [comment]
Thanks for checking with me!  I have fond memories of the Alpha architecture (Caml Special Light and OCaml version 1 were developed mainly on an Alpha workstation in the mid-90's), but I agree that its peculiar handling of byte-sized memory accesses is no longer of any interest.

**Comment by ghost** [comment]
Well I'm still using Alpha machines everyday... but I'm not running any OCaml code on them and I'm not really interested in bringing that backend back to life. :P
